src/Package/Impl/DataInspect/DataImport/ImportDataWindow.xaml.cs

```diff
@@ -309,11 +256,131 @@ class DataFramePreviewRowItem {
+            DeleteTempFile();
+            base.OnClosed(e);
+        }
+
+        private void PreviewFileContent(string file, int codePage) {
+            Encoding encoding = Encoding.GetEncoding(codePage);
+            string text = ReadFilePreview(file, encoding);
+            InputFilePreview.Text = text;
+        }
+
+        private Task ConvertToUtf8(string file, int codePage, int maxLines = Int32.MaxValue) {
+            DeleteTempFile();
+            bool reportProgress = maxLines == Int32.MaxValue;
+
+            // Convert file to UTF-8 so it can be properly imported
+            return Task.Run(async () => {
+                try {
+                    Encoding encoding = Encoding.GetEncoding(codePage);
+                    _tempFilePath = Path.GetTempFileName();
+                    int lineCount = 0;
+                    double progressValue = 0;
+
+                    using (var sr = new StreamReader(file, encoding, detectEncodingFromByteOrderMarks: true)) {
+                        if (reportProgress) {
+                            await StartReportProgress(Package.Resources.Converting);
+                        }
+                        long read = 0;
+                        using (var sw = new StreamWriter(_tempFilePath, append: false, encoding: new UTF8Encoding(encoderShouldEmitUTF8Identifier: false))) {
+                            string line;
+                            while (true) {
+                                line = sr.ReadLine();
+                                lineCount++;
+                                if (line == null || lineCount >= maxLines) {
+                                    break;
+                                }
+
+                                read += line.Length;
+                                sw.WriteLine(line);
+
+                                if (reportProgress) {
+                                    var newProgressValue = 90 * (double)read / (double)sr.BaseStream.Length;
+                                    if (newProgressValue - progressValue >= 10) {
+                                        progressValue = newProgressValue;
+                                        await ReportProgress(progressValue);
+                                    }
+                                }
+                            }
+                        }
+                    }
+                } catch (IOException ex) {
+                    VsAppShell.Current.DispatchOnUIThread(() => OnError(ex.Message));
+                }
+                if (reportProgress) {
+                    await ReportProgress(90);
+                }
+            });
+        }
+
+        private void DeleteTempFile() {
+            try {
+                if (!string.IsNullOrEmpty(_tempFilePath) && File.Exists(_tempFilePath)) {
+                    File.Delete(_tempFilePath);
+                }
+            } catch (UnauthorizedAccessException) { }
+        }
+
+        private async void DoDefaultAction() {
```

Comment:
Since this is `async void` and will crash if any exceptions leak, can you check that nothing in the code below can throw something potentially expected?